loggo/loggo.py

```diff
@@ -377,43 +376,47 @@ def _represent_return_value(self, response: Any, truncate: Optional[int] = 140)
 
         return '({})'.format(self._force_string_and_truncate(response, truncate, use_repr=True))
 
-    def _generate_log(self, where: str, returned: Any, formatters: Dict, safe_log_data: Dict[str, str]) -> None:
+    def _generate_log(self, msg: Optional[str], returned: Any, formatters: Dict, safe_log_data: Dict[str, str]) -> None:
         """
         generate message, level and log data for automated logs
 
-        where (str): 'pre'/'post'/'noreturn'/'error' --- the auto-log type
+        msg (str): the unformatted message
         returned (ANY): what the decorated callable returned
         formatters (dict): dict containing format strings needed for message
         safe_log_data (dict): dict of stringified, truncated, censored parameters
         """
+        # if the user turned off logs of this type, do nothing immediately
+        if not msg:
+            return
+
         # if errors not to be shown and this is an error, quit
-        if not self.allow_errors and where == 'error':
+        if not self.allow_errors and msg == self.errored:
             return
 
         # if state is stopped and not an error, quit
-        if self.stopped and where != 'error':
+        if self.stopped and msg != self.errored:
             return
 
         # do not log loggo, because why would you ever want that?
         if 'loggo.loggo' in formatters['call_signature']:
             return
 
-        # get the correct message
-        unformatted_message = FORMS[where]
-
         # return value for log message
-        if where == 'post':
+        if msg == self.returned:
             ret_str = self._represent_return_value(returned, truncate=None)
             formatters['return_value'] = ret_str
             formatters['return_type'] = type(returned).__name__
 
         # if what is 'returned' is an exception, get the error formatters
-        if where == 'error':
-            formatters['error_type'] = type(returned).__name__
-            formatters['error_string'] = str(returned)
+        if msg == self.errored:
+            formatters['exception_type'] = type(returned).__name__
+            formatters['exception_msg'] = str(returned)
+            formatters['level'] = self.error_level
+        else:
+            formatters['level'] = 20
```

Comment:
I wouldnt hard code 20 here. The value 20 has a connection to line 431 `self.info`. I would make make the integer a constant value defined only once, and the use it here, and on line 431 by doing `self.log(level...`. Also I think `logging.INFO` is more descriptive than `20`.

Reply:
yeah, fair enough, though i think we plan to also address this in more detail in a later MR as well

Reply:
have changed to use `logging.INFO`, and in `__init__` too, so now by default loggo only ever logs at info level unless user chooses otherwise via config or via e.g. `self.warning`. if you want anything more changes to error level, it's beyond scope of this PR

Reply:
Yeah I was hoping for the same value not to be hard coded on multiple lines. We still have logging.INFO here and self.info() call below, when we could get the level from a constant.